unit_tests/test_socket_connection.py

```diff
@@ -245,8 +246,9 @@ def receive_until(self, expected):
                         if not self.stay_silent:
                             self.server_socket.sendto(self.data_to_send, addr)
                         break
-                except socket.timeout:
-                    break
+                except Exception as e:
+                    self.exception = e
+                    raise
```

Comment:
I noticed that there was an exception while executing the `sendto()` part the whole time which remained undetected because pytest only checks for exceptions in the main thread. Hence this workaround where we need to check if `exception` is None in every test.
If anyone knows a better solution to make pytest thread aware, please let us know.

Reply:
Will be dealt with in another PR.